### PR TITLE
Enable zarr backend testing in data tests [1]

### PR DESCRIPTION
**Pull request by h-mayorquin**
This enables zarr testing for running conversions with backend specified as a string (still missing the building a backend configuration part)

It also makes some metadata checks proper tests on their own which has two advantages:
* They will fail on their own and be identified as such by pytest
* Can be parallelized with xdist.

### CODE REVIEW

File: src/neuroconv/tools/testing/data_interface_mixins.py [hunk 1]
```diff
@@ -223,6 +217,26 @@ def run_custom_checks(self):
         """Override this in child classes to inject additional custom checks."""
         pass
 
+    @pytest.mark.parametrize("backend", ["hdf5", "zarr"])
+    def test_run_conversion_with_backend(self, setup_interface, tmp_path, backend):
+
+        nwbfile_path = str(tmp_path / f"conversion_with_backend{backend}_{self.test_name}.nwb")
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/testing/data_interface_mixins.py:223]:
```suggestion
        nwbfile_path = str(tmp_path / f"conversion_with_backend-{backend}_{self.test_name}.nwb")
```

Comment (h-mayorquin) [reply src/neuroconv/tools/testing/data_interface_mixins.py:223]: you prefer `-` over `_`?

Comment (CodyCBakerPhD) [reply src/neuroconv/tools/testing/data_interface_mixins.py:223]: whichever you prefer

I kind of tend to mirror what DANDI does and `_` separate thematically similar blocks, with `-` separation within a block

Comment (h-mayorquin) [reply src/neuroconv/tools/testing/data_interface_mixins.py:223]: I really don't have a strong preference, I will add the commit, just curios.

Comment (h-mayorquin) [reply src/neuroconv/tools/testing/data_interface_mixins.py]: Changed

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD OK, we got the point were the failures are actually related to zarr, check out the video test failure:

https://github.com/catalystneuro/neuroconv/actions/runs/10744684959/job/29802201096?pr=1056

Does it ring any bell? I will dig into it.

**Comment by h-mayorquin** [comment]
Note for myself to also add the documentation:

#1019

**Comment by CodyCBakerPhD** [comment]
Don't believe I've seen this one myself, but from the message the path to the Zarr array (file): https://github.com/catalystneuro/neuroconv/actions/runs/10744684959/job/29802201096?pr=1056#step:31:3426

is not a valid Windows path (should have replaced the `/` with `\\`?)

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD 
OK, this should be ready but it requires the fix on #1070 before (as does everything else).

The cause of the Windows error is here:

https://github.com/hdmf-dev/hdmf-zarr/issues/219

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD this is ready to go.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Coolio

**Pull request merged**